scripts/generate-error-types.tcl

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/tclsh
+#*
+#* SRT - Secure, Reliable, Transport
+#* Copyright (c) 2020 Haivision Systems Inc.
+#* 
+#* This Source Code Form is subject to the terms of the Mozilla Public
+#* License, v. 2.0. If a copy of the MPL was not distributed with this
+#* file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#* 
+#*/
+#
+#*****************************************************************************
+#written by
+#  Haivision Systems Inc.
+#*****************************************************************************
+
+set code_major {
+    UNKNOWN    -1
+    SUCCESS     0
+    SETUP       1
+    CONNECTION  2
+    SYSTEMRES   3
+    FILESYSTEM  4
+    NOTSUP      5
+    AGAIN       6
+    PEERERROR   7
+}
+
+set code_minor {
+    NONE             0
+    TIMEOUT          1
+    REJECTED         2
+    NORES            3
+    SECURITY         4
+    CLOSED           5
+
+    
+    CONNLOST         1
+    NOCONN           2
+    
+    THREAD           1
+    MEMORY           2
+    OBJECT           3
+    
+    SEEKGFAIL        1
+    READFAIL         2
+    SEEKPFAIL        3
+    WRITEFAIL        4
+    
+    ISBOUND          1
+    ISCONNECTED      2
+    INVAL            3
+    SIDINVAL         4
+    ISUNBOUND        5
+    NOLISTEN         6
+    ISRENDEZVOUS     7
+    ISRENDUNBOUND    8
+    INVALMSGAPI      9
+    INVALBUFFERAPI  10
+    BUSY            11
+    XSIZE           12
+    EIDINVAL        13
+    EEMPTY          14
+    
+    WRAVAIL          1
+    RDAVAIL          2
+    XMTIMEOUT        3
+    CONGESTION       4
+}
+
+
+set errortypes {
+
+	SUCCESS "Success" {
+		NONE      ""
+	}
+
+	SETUP "Connection setup failure" {
+		NONE      ""
+		TIMEOUT   "connection timed out"
+		REJECTED  "connection rejected"
+		NORES     "unable to create/configure SRT socket"
+		SECURITY  "aborted for security reasons"
+		CLOSED    "socket closed during operation"
+	}
+
+    CONNECTION "" {
+		NONE     ""
+		CONNLOST "Connection was broken"
+		NOCONN   "Connection does not exist"
+	}
+
+    SYSTEMRES "System resource failure" {
+		NONE   ""
+		THREAD "unable to create new threads"
+		MEMORY "unable to allocate buffers"
+		OBJECT "unable to allocate a system object"
+	}
+
+    FILESYSTEM "File system failure" {
+		NONE      ""
+		SEEKGFAIL "cannot seek read position"
+		READFAIL  "failure in read"
+		SEEKPFAIL "cannot seek write position"
+		WRITEFAIL "failure in write"
+	}
+
+    NOTSUP "Operation not supported" {
+		NONE           ""
+        ISBOUND        "Cannot do this operation on a BOUND socket"
+        ISCONNECTED    "Cannot do this operation on a CONNECTED socket"
+        INVAL          "Bad parameters"
```

Comment:
There are some alignment issues (mixed tabs and spaces) here and in the file in general.